trl/trainer/training_configs.py

```diff
@@ -41,3 +41,33 @@ class RewardConfig(TrainingArguments):
     """If True, use gradient checkpointing to save memory at the expense of slower backward pass."""
     gradient_checkpointing_kwargs: Optional[dict] = None
     """Keyword arguments to pass to the gradient checkpointing function."""
+
+
+@dataclass
+class KTOConfig(TrainingArguments):
```

Comment:
Thanks for creating a separate `KTOConfig` class! 

I originally placed the `RewardConfig` under this general-purpose module, but given the proliferation of new trainers, I am now thinking that it might make sense to just have dedicated modules per trainer, i.e. something like:

```
- kto_config.py
- kto_trainer.py
- reward_config.py
- reward_trainer.py
```

This way it's a bit easier to find the relevant code as a developer / power user. WDYT @lvwerra @younesbelkada ?

Reply:
makes sense to me !

Reply:
PR here: https://github.com/huggingface/trl/pull/1221